e2e/v3/jobs3/jobs3.go

```diff
@@ -315,6 +316,10 @@ EVAL:
 		}
 	}
 
+	if !sub.noCleanup {
+		sub.t.Cleanup(sub.cleanup)
+	}
```

Comment:
I think this hits the same problem you described in the PR. If the eval API returns a fatal condition, we'll never cleanup. So I think you want to move this right underneath the `Register` API call, right after checking if you got an error from that.

Reply:
Ah yeap, I got myopic about the failures I have seen, which have been "deployment wasn't happy fast enough" and neglected earlier possible errors.